uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/shared/models/instances/leave-request-instance_test.js

```diff
@@ -575,5 +575,25 @@ define([
         return $q.resolve(value);
       }
     });
+
+    describe('toAPI()', function () {
```

Comment:
`toAPI()` is defined in the parent `Instance` service, you can see that `LeaveRequestInstance` uses it here
```js
toAPIData = leaveRequest.toAPI();
```
`toApi()` in turns uses the `toAPIFilter` in which you have the transformations. The reason that `toAPIFilter` is exposed in the object interface is that allows each instance to extend the original `toAPIFilter` defined in `Instance`
```js
/**
 * Function that filters data meant to be sent to the API, being
 * called in as the iteratee of a _.transform() call
 *
 * Can be overriden by children api for custom filtering
 *
 * @param {object} result - The accumulator object
 * @param {string} key - The property name
 */
toAPIFilter: function (result, __, key) {
  result[key] = this[key];
}
```